src/test/java/org/opendevstack/provision/services/MailAdapterTest.java

```diff
@@ -14,70 +14,54 @@
 
 package org.opendevstack.provision.services;
 
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.*;
 
-import javax.mail.internet.MimeMessage;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
 import org.opendevstack.provision.authentication.crowd.CrowdAuthenticationManager;
 import org.opendevstack.provision.model.OpenProjectData;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.mail.javamail.JavaMailSender;
-import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.ActiveProfiles;
+import org.springframework.mail.javamail.MimeMessagePreparator;
 
-@SpringBootTest
-@DirtiesContext
-@ActiveProfiles("crowd")
+@ExtendWith(MockitoExtension.class)
 public class MailAdapterTest {
 
-  @Autowired private MailAdapter mailAdapter;
+  private MailAdapter mailAdapter;
 
-  @MockBean private CrowdAuthenticationManager crowdAuthenticationManager;
+  @Mock private CrowdAuthenticationManager crowdAuthenticationManager;
 
   @Mock private JavaMailSender mailSender;
 
   @BeforeEach
   public void setUp() {
+    mailAdapter = new MailAdapter(mailSender);
     mailAdapter.manager = crowdAuthenticationManager;
```

Comment:
Problem with this direct assignment to internal member of class is:
1. breaks historical and wide adopted java conventions of doing this with a setter. So devs do not expect this kind of implementation.  
2. I believe (not 100% sure) bytecode enhancer tools like asm decorates code based on getter and setter (due to limitation in the java reflextion functionality) Such tools are used by testing frameworks to instrument the code. If this rule is not followed then the tool might not be able to work as expected, which could lead to unexpected behaviour.
  
Therefore, can you please create a setter method for this?

Reply:
Hey! yes I did not change it because it did not yet wanna touch the MailAdapter class. The best way here would be to also inject this manager service via constructor injection as it is done with mailSender. I can do this later.

Reply:
ok, let us improve the `MailAdapter` in a later change.